pkg/apis/httpEndpoint/v1alpha1/types.go

```diff
@@ -51,11 +61,61 @@ func (h HTTPEndpoint) NameValuePairs() []common.NameValuePair {
 	return h.Spec.Headers
 }
 
+// nil returns a bool indicating if a client key has a secret reference
+func (h HTTPEndpoint) HasTLSClientKeySecret() bool {
+	return h.Spec.ClientKey != nil && h.Spec.ClientKey.SecretKeyRef != nil && h.Spec.ClientKey.SecretKeyRef.Name != ""
+}
+
+// HasTLSClientCertSecret returns a bool indicating if a client cert has a secret reference
+func (h HTTPEndpoint) HasTLSClientCertSecret() bool {
+	return h.Spec.ClientCert != nil && h.Spec.ClientCert.SecretKeyRef != nil && h.Spec.ClientCert.SecretKeyRef.Name != ""
+}
+
+// HasTLSRootCASecret returns a bool indicating if a root cert has a secret reference
+func (h HTTPEndpoint) HasTLSRootCASecret() bool {
+	return h.Spec.RootCA != nil && h.Spec.RootCA.SecretKeyRef != nil && h.Spec.RootCA.SecretKeyRef.Name != ""
+}
```

Comment:
I think it would be better to return true when this string is empty. An empty string signals that the API field was in fact defined and therefore should be processed. Yes, this will likely result in an error downstream because the name is empty, however that error is correct- that is what the user defined. Not returning true here means that the field will be silently ignored, even when defined.

Reply:
Disagree, these methods serve a specific purpose which isn't to signal if the field was defined, but rather allow the caller to derive a conclusion based on whether or not a secret has been defined to be processed for fetching (hence secret.name not empty) or if a value was set.

Reply:
Shouldn't the end result of this conclusion not be that the caller should error out (eventually to the user) that they have defined a secret reference, but that reference has failed validation- rather than behaving as though the `secretKeyRef` was not defined at all.

Reply:
No. Checking if `secretKeyRef` was defined can be done via performing a simple nil check against the field.  This helps callers avoid repeating convoluted code when needing to make a decision whether or not a secret should be extracted, which goes further than the simple nil check.